asymmetric.go

```diff
@@ -282,10 +283,31 @@ func (ctx rsaDecrypterSigner) signPayload(payload []byte, alg SignatureAlgorithm
 
 	var out []byte
 	var err error
+	var goVer *version.Version
+	var constraints version.Constraints
 
 	switch alg {
 	case RS256, RS384, RS512:
-		out, err = rsa.SignPKCS1v15(RandReader, ctx.privateKey, hash, hashed)
+		// Allow relying parties to use PKCS1v15 signing algorithms on versions
+		// of golang < 1.20 while still satisfying the rsa.SignPKCS1v15 random reader
+		// parameter deprecation.
+		// TODO(@pgporada) Remove this version checking
+		goVer, err = goVersionCheck()
+		if err != nil {
+			return Signature{}, fmt.Errorf("%v", err)
+		}
+		constraints, err = version.NewConstraint("< 1.20")
+		if err != nil {
+			return Signature{}, fmt.Errorf("%v", err)
+		}
+		if constraints.Check(goVer) {
```

Comment:
I don't think we should be doing this in the production code. There's nothing _wrong_ with calling SignPKCSv15 with a RandReader -- the parameter will be ignored, and that's just fine. If the go team decides to actually remove the parameter in some future version, we'll have plenty of warning and time to update then.